File: server/server/models/usermaster.py
from sqlalchemy import (
    Column,
    Integer,
    Text,
    ForeignKey,
)
import hashlib
import six
from .meta import Base


class UserMaster(Base):
    __tablename__ = 'user_master'
    id = Column(Integer, primary_key=True)
    name = Column(Text)
    role = Column(Integer)
    title = Column(Integer)
    designation = Column(Text)
    login = Column(Text)
    password = Column(Text)
    status = Column(Integer)

    def __init__(self, login, name, role, title, designation, status):
        self.name = name
        self.login = login
        self.role = role
        self.title = title
        self.designation = designation
        self.status = status

    @classmethod
    def get_users(cls, DBSession):
        return DBSession.query(UserMaster).all()

    @classmethod
    def check_user(cls, DBSession, name):
        return  DBSession.query(UserMaster).filter_by(name=name).first()

    @classmethod
    def get_user(cls, DBSession, id):
        return  DBSession.query(UserMaster).filter_by(id=id).first()

    @classmethod
    def delete_user(cls, DBSession, id):
        DBSession.query(UserMaster).filter_by(id=id).delete()
        return true

    @classmethod
    def check_user_vo(cls, DBSession, login):
        return DBSession.query(UserMaster).filter_by(login=login).first()

    def set_password(self, password):
        self.password = _sha512(password)

    def check_password(self, password):
        return self.password == _sha512(password)

    @classmethod
    def by_login(cls, DBSession, login):
        return DBSession.query(UserMaster).filter_by(login=login).first()


class UserFingerPrintMap(Base):
    __tablename__ = 'user_fingerprint'
    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey('user_master.id'))
    file_name = Column(Text)

    def __init__(self, user_id, file_name):
        self.user_id = user_id
        self.file_name = file_name


def _sha512(text):
    sha = hashlib.sha512()
    sha.update(six.b(text))
    return sha.hexdigest()

File: client/src/app/user-master/model/UserMaster.ts
import {Roles, Status, Titles} from '../../constant';

export class UserMaster {
  id: number;
  name: string;
  title: number;
  role: number;
  userId: string;
  password: string;
  confirmPassword: string;
  status = 1;
  designation: string;

  static fromJson(data: Map<string, any>): UserMaster {
    const u: UserMaster = new UserMaster();
    u.id = data['id'];
    u.userId = data['login']
    u.name = data['name'];
    u.title = data['title'];
    u.role = data['role'];
    u.status = data['status'];
    u.designation = data['designation'];
    return u;
  }

  toJSON(): Map<string, any> {
    const data: Map<string, any> = new Map<string, any>();
    data['id'] = this.id;
    data['userId'] = this.userId;
    data['name'] = this.name;
    data['title'] = this.title;
    data['role'] = this.role;
    data['password'] = this.password;
    data['status'] = this.status;
    data['designation'] = this.designation;
    data['confirm-password'] = this.confirmPassword;
    return data;
  }
}
